Project/Src/Gui/GutterMargin.cs

```diff
@@ -42,7 +42,7 @@ public GutterMargin(TextArea textArea) : base(textArea)
         public override Cursor Cursor => RightLeftCursor;
 
         public override int Width
-            => textArea.TextView.WideSpaceWidth*Math.Max(4, (int)Math.Log10(textArea.Document.TotalNumberOfLines) + 4);
+            => textArea.TextView.WideSpaceWidth * (2 * (int)Math.Log10(textArea.Document.TotalNumberOfLines) + 3) / 2;
```

Comment:
```suggestion
            => textArea.TextView.WideSpaceWidth * ((int)Math.Log10(Math.Max(1, textArea.Document.TotalNumberOfLines)) * 2 + 3) / 2;
```
if we cannot rule out 0 lines.

Reply:
I think it is OK without Math.Min now.
For me WideSpaceWidth so width is 3 pixels, which looks OK. You may briefly see 0 width flashing by when selecting a new commit with e.g. a folder.
An empty file is listed as 1line, 0 should be explicitly cleared.